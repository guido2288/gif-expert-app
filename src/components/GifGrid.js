import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFechGif';


export const GifGrid = ({ category} ) => {

    const { data, loading} = useFetchGifs( category );
    


    return (
        <>
            <h3 className='card animate__animated animate__fadeIn'>{ category }</h3>

            { loading && <p>Loading...</p> }

            <div className='card-grid'>      
                {
                    data.map( img => (
                        <GifGridItem  
                            key= { img.id }
                            {...img}/>
                    ))
                }     
            </div>
        </>
    )
}
