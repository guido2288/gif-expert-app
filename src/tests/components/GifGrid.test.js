import React from 'react'
import '@testing-library/jest-dom'

import {shallow} from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFechGif';
jest.mock('../../hooks/useFechGif') //esto es para fingir cualquier llamada a ese archivo y suponer o controlar la info que va a responder

describe('Pruebas en <GifGrid />', () => {
    
    const category = 'One Punch';
    

    test('debe mostrarse correctamente ', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });//esto hace que cuando se llame el useFetch va a regresar el valor que le declaro

        const wrapped = shallow( <GifGrid category= { category } /> )
        expect(wrapped).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se carga la imagen useFetchGifs ', () => {


        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier Cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        const wrapped = shallow( <GifGrid category= { category } /> )

        expect( wrapped.find('GifGridItem').length).toBe(gifs.length)//se toma el nombre del elemento GifGridItem
        
        
    })
    
    

})
