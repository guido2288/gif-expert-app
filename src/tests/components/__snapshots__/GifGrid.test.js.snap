// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrarse correctamente  1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
